docs/architecture.puml

```diff
@@ -0,0 +1,63 @@
+@startuml
```

Comment:
@B2o5T any idea if we can add some magic to display this kind of file (similar to what we have with mermaid)? 

In the same context, @Enrico2 any plan to move this to be in mermaid format?